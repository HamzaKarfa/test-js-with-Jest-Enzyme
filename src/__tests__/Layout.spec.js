import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Layout from '../components/Layout';
import Home from '../components/Home';
import '../setUpTests';



//Créer un mock de Home
jest.mock("../components/Home", () => 'Home' )

describe('Layout snapshot',()=>{
    let layout
    beforeEach(() => {
        // Clear all instances and calls to constructor and all methods:
        layout = shallow(
            <Layout />
          )
      });
      
    it('It renders without crashing', () => {
        //Créer le expect
        expect(
            renderer.create(<Layout />).toJSON()
        ).toMatchSnapshot()
    });

    it('It renders Home', () => {
        //Créer snapshot*
        expect(layout.find('Home')).toMatchSnapshot()
    });
    
});
