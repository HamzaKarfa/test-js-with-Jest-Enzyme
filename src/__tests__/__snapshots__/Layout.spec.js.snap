// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot It renders Home 1`] = `
<Home
  fetchRandomNumber={[Function]}
/>
`;

exports[`Layout snapshot It renders without crashing 1`] = `
<div>
  <h1>
    Basic Calculator
  </h1>
  <Home
    fetchRandomNumber={[Function]}
  />
</div>
`;
